### ISSUE TITLE
How should swagger-codegen deal with multiple inline objects within allOf: ?

### ISSUE THREAD

**Issue by jeff9finger** [swagger-api/swagger-codegen#4231]
##### Description
Currently, swagger-codegen only recognizes that last inline object defined in an allOf: array. The OpenAPI Specification is silent on this issue and leaves it to the tools to decide.

https://github.com/OAI/OpenAPI-Specification/issues/848

I am writing this issue so that this issue can be defined in swagger-codegen.

##### Swagger-codegen version

2.2.2-SNAPSHOT

##### Swagger declaration file content or url

See issue # 4226 for example.

##### Suggest a Fix

Specify a constraint that only one inline object can be created within a model definition.
OR
Handle it properly. If this option is chosen, how are discriminators dealt with in the inline objects? What other issues will come up when this is addressed?

IIUC, 2.2.3 addresses inheritance, interfaces and composition in a more complete way, right? So maybe for 2.2.2, the constraint is the best option.

**Comment by webron** [comment]
Again, what is the value of having multiple discriminators? Just trying to understand the use case.

**Comment by jeff9finger** [comment]
I don't see value in it. But currently, swagger-codegen simply ignores all of them except the last one. I just want to clarify this,  because if the specification allows it and someone does try this for some reason that we can't anticipate,  the generated code is incorrect and not what is expected.

This came up for me in trying to set a discriminator on the online object.  The discriminator field is not set in the ModelImpl for inline objects and I am trying to determine if simply setting it on a sub object is correct, or of other inline objects would need to handle this as well.

So, I'm simply looking for clarity on this issue as it is currently undefined (and arguably incorrect if someone does specify more than one inline object) and could lead to confustion

**Comment by webron** [comment]
I tend to disagree. The spec is not fault-proof, and it doesn't mean that the tools can and should deal with all cases. We should support features as they are requested by users if not tackled, but it cannot be without reason. Yes, inline models for `allOf` should be supported because that's a simple use case. Multiple `discriminator` fields is borderline abuse of the spec.

I also don't expect the codegen to produce 'usable' code for this:
```yaml
allOf:
  - type: object
    properties:
      foo:
        type: string
  - type: object
    properties:
      foo:
        type: integer
```

Just because you can write it, doesn't mean it makes sense.

There may very well be a reason behind using multiple `discriminator`s. I'd like to learn of that use case so if needed, we can even add better support in the spec (future versions, of course).

**Comment by jeff9finger** [comment]
I think we are agreeing... :-) 

When you say "multiple discriminators" are you referring to something like:
```
allOf:
  - type: object
    discriminator: foo_type
    properties:
      foo:
        type: string
      foo_type:
        type: string
  - type: object
    discriminator: bar_type
    properties:
      foo:
        type: integer
      bar_type:
        type: string
```

I don't expect the above spec to produce usable code either and agree that it  "is borderline abuse of the spec".

I do think the following is a good use case:
```
definitions:

  Object:
    allOf:
      - $ref: '#/definitions/SomeOtherObject'
      - type: object
        discriminator: object_type
        required:
          - object_type
        properties:          
          object_type:
            type: string
          name:
            type: string
```

And this is the issue that I am trying to resolve. As part of that effort, I found that there was nothing that stated that using "multiple discriminators" was leaning toward an abuse of the spec or that some of the inline objects would be ignored or anything.

My reason for opening this issue is simply to ask the community what the expected behavior is and to specify that as a constraint or a warning, at least for now. That way if someone does want this in the future, it is clearly stated what the limitation is.

I may be in the minority here, but I think it would be valuable to have a warning or documentation or something  that specifies what the current behavior is.
